AppDB/backup_data.py

```diff
@@ -0,0 +1,367 @@
+""" This process performs a backup of all the application entities for the given
+app ID to the local filesystem.
+"""
+import argparse
+import cPickle
+import logging
+import os
+import random
+import re
+import shutil
+import sys
+import threading
```

Comment:
Instead of threading you may want to use the multiprocessing module which will use all the cores on the system.